### PR TITLE
Adding resource policies to openapi

### PR DESCRIPTION
**Pull request by Bento007**
- Supporting CRUD for resource policies
- Supporting CRUD for resource actions
- rename resource_policy to policy_document which is closer to how it's actually stored in cloud directory.
- changed how list_policies and list_ids are returned as a dict. This is how the rest of fusillade lists objects.

### CODE REVIEW

File: fusillade/directory/resource.py [hunk 1]
```diff
@@ -319,6 +319,7 @@ def get_policy(self, policy_name):
             attrs = dict([(attr['Key']['Name'], attr['Value'].popitem()[1]) for attr in resp['Attributes']])
         except cd_client.exceptions.ResourceNotFoundException:
             raise FusilladeNotFoundException(f"{self.get_policy_path(policy_name)} does not exist.")
+        attrs['policy_document'] = json.loads(attrs['policy_document'])
```
Comment (chmreid) [inline fusillade/directory/resource.py]: Just an observation, related to changing the policy document format from dictionary to JSON string. AWS also has some inconsistencies in how boto returns policy documents (some API endpoints return policies as a dict, others as a JSON string). May be useful to clarify somewhere (I don't really know where...?) that policy documents are stored as stringified JSON.

File: fusillade-api.yml [hunk 2]
```diff
@@ -1122,13 +1359,19 @@ components:
       type: array
       items:
         type: string
+    policy_type:
+      type: string
+      enum: ['ResourcePolicy', 'IAMPolicy']
```
Comment (chmreid) [inline fusillade-api.yml]: Is the IAMPolicy type TBA? Would it be better to modify this to `['ResourcePolicy']` only until IAMPolicy policies are added?

File: tests/test_resource_api.py [hunk 3]
```diff
@@ -80,6 +81,146 @@ def test_get_resource(self):
             self.app.post(f'/v1/resource/tr{i}', data=json.dumps({'actions': ['tr:action1']}), headers=admin_headers)
         self._test_paging('/v1/resource', admin_headers, 10, 'resources')
 
+    def test_get_resource_policy(self):
+        """Pages of resource are retrieved when using the get resource API"""
+        resp = self.app.post(f'/v1/resource/trp', data=json.dumps({'actions': ['trp:action1']}),
```
[the remark this answers is not in the document]
Comment (Bento007) [reply tests/test_resource_api.py]: I got tired of typing. I'll leave it for now.

File: fusillade-api.yml (hunk 2 above)
Comment (Bento007) [reply fusillade-api.yml]: It's already used in other places in the code, such as roles, users, and groups. A resource can potential have IAM policies attached to it as well. I'll remove it until it's used.

File: fusillade/directory/resource.py (hunk 1 above)
Comment (Bento007) [reply fusillade/directory/resource.py]: The openapi schema documents the policy_document format. In a future PR I will clean up the openapi spec to make sure everything is consistent.

File: fusillade-api.yml [hunk 4]
```diff
@@ -1038,6 +1038,250 @@ paths:
...
+        401:
+          $ref: '#/components/responses/401'
+        403:
+          $ref: '#/components/responses/403'
+        404:
+          $ref: '#/components/responses/404'
+        default:
+          description: Unexpected error.
+    post:
+      summary: Add a new resource policy
```
[the remark this answers is not in the document]
Comment (Bento007) [reply fusillade-api.yml]: I expanded on the summaries.

[the remark this answers is not in the document]
Comment (Bento007) [reply fusillade-api.yml]: "Modified an existing resource policy" ->"Modify an existing resource policy for the specified resource type"

[the remark this answers is not in the document]
Comment (Bento007) [reply fusillade-api.yml]: You comment made it clear to me that my description was not sufficient. Thank you.

**Pull request merged**